### PR TITLE
fix: [lw-11877]: force Nami-mode users to switch to Lace delegating from dapp popup

### PR DESCRIPTION
**Pull request by vetalcore**
# Checklist

- [x] JIRA - [LW-11877](https://input-output.atlassian.net/browse/LW-11877)
- [ ] Proper tests implemented
- [x] Screenshots added.

---

## Proposed solution

Explain how does this PR solves the problem stated in JIRA ticket.
You can also enumerate different alternatives considered while approaching this task.

## Testing

Describe here, how the new implementation can be tested.
Provide link or briefly describe User Acceptance Criteria/Tests that need to be met

## Screenshots

Screenshot 2024-11-22 at 19 40 22

[LW-11877]: https://input-output.atlassian.net/browse/LW-11877?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by pczeglik-iohk** [comment]
# Allure Report
[`allure-report-publisher`](https://github.com/andrcuns/allure-report-publisher) generated test report!



**processReports**: ❌ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/smoke/linux/chrome/7479/11977921910/index.html) for [e7586173](https://github.com/input-output-hk/lace/pull/1559/commits/e758617381a3be5501421e94d18fc6c873537cd2)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 31     | 1      | 4       | 0     | 36    | ❌     |

File: apps/browser-extension-wallet/src/views/nami-mode/NamiDappConnectorView.tsx [hunk 1]
```diff
@@ -196,14 +198,35 @@ export const NamiDappConnectorView = withDappContext((): React.ReactElement => {
...
   const handleResolver = useHandleResolver();
 
+  useEffect(() => {
+    getBackgroundStorage()
+      .then((storage) => setNamiMigration(storage.namiMigration))
+      .catch(console.error);
+  }, []);
+
+  const switchWalletMode = useCallback(async () => {
+    const mode = namiMigration?.mode === 'lace' ? 'nami' : 'lace';
```
Comment (mirceahasegan) [inline apps/browser-extension-wallet/src/views/nami-mode/NamiDappConnectorView.tsx:208]: Should `undefined` mode be equivalent to `lace` mode? Reading it as: no migration done, so still in lace mode.
```diff
-    const mode = namiMigration?.mode === 'lace' ? 'nami' : 'lace';
+    const mode = namiMigration?.mode === 'nami' ? 'lace' : 'nami';
```
Also, this switch seems to be available only in nami* components. Is this check needed?

File: packages/nami/src/ui/app/pages/dapp-connector/signTx.tsx [hunk 2]
```diff
@@ -311,7 +314,40 @@ export const SignTx = ({
               {dappInfo?.url.split('//')[1]}
             </Text>
           </Box>
-          <Box h="8" />
+          <Box h="4" />
```
[the remark this answers is not in the document]
Comment (vetalcore) [reply packages/nami/src/ui/app/pages/dapp-connector/signTx.tsx]: addressed here 43efeb6c25b36a01c619b2fe17681bf3ad14d265

File: packages/nami/src/ui/app/pages/dapp-connector/signTxUtil.ts [hunk 3]
```diff
@@ -390,3 +391,14 @@ const toAssetInfo = (
   }
   return toAsset(assetInfo, BigInt(cardanoAsset.quantity));
 };
+
+export const txHasGovernanceFields = ({
+  body: { certificates, votingProcedures, proposalProcedures },
+}: Wallet.Cardano.Tx) =>
+  certificates?.some(certificate =>
+    Object.values(ConwayEraCertificatesTypes).includes(
```
Comment (mirceahasegan) [inline packages/nami/src/ui/app/pages/dapp-connector/signTxUtil.ts:399]: `ConwayEraCertificatesTypes` also contains some non-governance certificates: `Registration`, `Unregistration` and `StakeRegistrationDelegation`.
Please filter them out before the check.

Comment (vetalcore) [reply packages/nami/src/ui/app/pages/dapp-connector/signTxUtil.ts]: addressed here 43794a802c314f927f0ee4ee6a2af65671d920cd

**Review by mirceahasegan — APPROVED** [review]
Great work @vetalcore  🥇

**Pull request merged**